src/middlewares/auth.middleware.ts

```diff
@@ -50,11 +57,33 @@ export const restrictTo = (...roles: any) => {
   };
 };
 
-// export const isVerfied = (req: Request, res: Response, next: NextFunction) => {
-//   if (!req.user.verfied) {
-//     return res.status(403).json({
-//       message: "You are not authorized to perform this action",
-//     });
-//   }
-//   next();
-// };
\ No newline at end of file
+export const verifyToken = (req: Request, res: Response, next: NextFunction) => {
+  const authHeader = req.header('Authorization');
+  const token = authHeader && authHeader.split(' ')[1];
+
+  if (!token) {
+    return res.status(401).json({
+      status: "error",
+      message: "Access denied. No token provided.",
+    });
+  }
+
+  if (isBlacklisted(token)) {
+    return res.status(401).json({
```

Comment:
We already have this middleware all you need to do is to check in protectRoute if token is blacklisted

Reply:
The changes resolved bruh